src/main/java/org/cloudfoundry/promregator/cfaccessor/ReactiveCFAccessorImpl.java

```diff
@@ -220,6 +229,18 @@ private void constructCloudFoundryClient() throws ConfigurationException {
 			
 			log.info("Target CF platform is running on API version {}", apiVersion);
 		}
+
+		// Ensures v3 API exists. The CF Java Client does not yet implement the info endpoint for V3, so we do it manually.
+		JsonNode v3Info = new ReactorInfoV3(this.cloudFoundryClient.getConnectionContext(), this.cloudFoundryClient.getRootV3(),
+											this.cloudFoundryClient.getTokenProvider(), this.cloudFoundryClient.getRequestTags())
+			.get().onErrorReturn(JsonNodeFactory.instance.nullNode()).block();
+
+		if (v3Info == null || v3Info.isNull()) {
+			log.warn("unable to get v3 info endpoint of CF platform, some features will not work as expected");
```

Comment:
That may be a little weak.. I see your intention, but what should the common user of Promregator do now? He will just be scared and unsure what the system does and how it behaves. 

My initial ideas at https://github.com/promregator/promregator/pull/196#discussion_r609990524 had gone into the direction of implementing a https://github.com/promregator/promregator/blob/576b5f894b883109270aff67146ab85d120b8cc2/src/main/java/org/cloudfoundry/promregator/config/validations/ConfigurationValidation.java.  As https://github.com/promregator/promregator/blob/576b5f894b883109270aff67146ab85d120b8cc2/src/main/java/org/cloudfoundry/promregator/config/ConfigurationValidations.java is a spring bean, it should be possible to also `Autowire` the `ReactiveCFAccessor` whose interface could be extended by your `v3Enabled`. I expect it won't be the last time that we will run into the situation that "some feature" won't run without v3 available.

Reply:
I've implemented a validator. I did some refactoring to the existing ones to just be beans so it's easier to extend these validators to use other beans in the future.